### PR TITLE
Implemented delete Car functionality

### PR DESCRIPTION
**Pull request by GordonOtieno**
- Implement fetch all cars and display and display on nice UI.
- Create a delete car form and style.
- Implement delete functionality to the api.
- Solve linters errors.
- Update router to include delete functionality
- update backend to include cors., so pull to update before testing.

### CODE REVIEW

**Comment by GordonOtieno** [comment]
Please pull to update backend dev branch. There is a slight change that is necessary for working of this code..

File: src/components/deletecar/RemoveCar.jsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        mode: 'no-cors',
+      },
+      body: JSON.stringify({
+        id: carId,
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          console.log('Resource deleted successfully');
+          // alert('Car successfully deleted');
```
Comment (ouasamine) [inline src/components/deletecar/RemoveCar.jsx:36]: - Please make sure to remove these `console.log`s.

File: src/components/deletecar/RemoveCar.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          setShowModal({
+            alert: true,
+            message: `Yaaaay! You successfully deleted ${carName}`,
+            type: 'success',
+          });
+        } else {
+          console.error('Error deleting resource');
+        }
```
Comment (ouasamine) [inline src/components/deletecar/RemoveCar.jsx:44]: - Please make sure to remove these `console.log`s.

File: src/components/deletecar/RemoveCar.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      })
+      .catch((error) => {
+        console.error('Error deleting resource:', error);
+        setShowModal({
```
Comment (ouasamine) [inline src/components/deletecar/RemoveCar.jsx:48]: - Please make sure to remove these `console.log`s.

**Review by ouasamine — APPROVED** [review]
Hi @GordonOtieno Nice work 👏🏽  the PR is good to be merged, I have some observations and it would be nice if you could implement them before merging:
- Currently, when I click on the delete button, the car is deleted successfully from the database, but it doesn't disappear from the page nor do I get any success message, so a normal user would have no clue if the operation has been successful or not and would probably click on the button again, please try to implement a function to refresh the content of the page after the delete.
- The modal that you implemented was a very good idea, but I think its design could be further improved, currently it's transparent, and I don't think that looks good enough 🔽 
 
popup

**Pull request merged**